platform/kube/admit/admit.go

```
@@ -0,0 +1,248 @@
+	return &Admit{
+		descriptor:         descriptor,
+		hookConfigName:     hookConfigName,
+		hookName:           hookName,
+		serviceName:        serviceName,
+		serviceNamespace:   serviceNamespace,
+		caBundle:           caBundle,
+		validateNamespaces: validateNamespaces,
+	}
+}
+
+// Unregister registers the external admission webhook
+func (a *Admit) Unregister(client admissionClient.ExternalAdmissionHookConfigurationInterface) error {
+	return client.Delete(a.hookConfigName, nil)
+}
+
+// Register registers the external admission webhook for pilot
+// configuration types.
+func (a *Admit) Register(client admissionClient.ExternalAdmissionHookConfigurationInterface) error {
+	var resources []string
+	for _, schema := range a.descriptor {
+		resources = append(resources, schema.Plural)
+	}
+
+	webhook := &admissionregistrationv1alpha1.ExternalAdmissionHookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: a.hookConfigName,
+		},
+		ExternalAdmissionHooks: []admissionregistrationv1alpha1.ExternalAdmissionHook{
+			{
+				Name: a.hookName,
+				Rules: []admissionregistrationv1alpha1.RuleWithOperations{{
+					Operations: []admissionregistrationv1alpha1.OperationType{
+						admissionregistrationv1alpha1.Create,
+						admissionregistrationv1alpha1.Update,
+					},
+					Rule: admissionregistrationv1alpha1.Rule{
+						APIGroups:   []string{model.IstioAPIGroup},
+						APIVersions: []string{model.IstioAPIVersion},
+						Resources:   resources,
+					},
+				}},
+				ClientConfig: admissionregistrationv1alpha1.AdmissionHookClientConfig{
+					Service: admissionregistrationv1alpha1.ServiceReference{
+						Namespace: a.serviceNamespace,
+						Name:      a.serviceName,
+					},
+					CABundle: a.caBundle,
+				},
+			},
+		},
+	}
+	client.Delete(webhook.Name, nil) // nolint: errcheck
```

might be worth logging even if ignoring the error, for debugging purposes